tao_compiler/mlir/disc/transforms/disc_hlo_legalize_to_lhlo.cc

```diff
@@ -209,6 +209,50 @@ struct TieShapeOpConverter : public BaseOpConversion<TieShapeOp> {
   }
 };
 
+struct SparseReshapeOpConverter
+    : public BaseOpConversion<mhlo_disc::SparseReshapeOp> {
+ public:
+  using BaseOpConversion<mhlo_disc::SparseReshapeOp>::BaseOpConversion;
+
+  LogicalResult matchAndRewrite(
+      mhlo_disc::SparseReshapeOp op, OpAdaptor adaptor,
+      ConversionPatternRewriter& rewriter) const override {
+    auto loc = op.getLoc();
+    auto operands = adaptor.getOperands();
+
+    SmallVector<Value, 5> buffer_args;
+    // input indices
+    // input shape
+    // new shape
+    for (Value memref : operands) {
+      buffer_args.push_back(memref);
+    }
+
+    Value num_values = rewriter.create<memref::DimOp>(loc, operands[0], 0);
+    Value new_rank = rewriter.create<memref::DimOp>(loc, operands[2], 0);
+    SmallVector<Value, 2> alloc_operands({num_values, new_rank});
+    auto alloc = rewriter.create<memref::AllocOp>(
```

Comment:
better to call the shape inference interface implementation of such op, instead of explicit doing shape interface here.

Reply:
done